apps/dsp-app/src/app/user/user-form/user-form.component.ts

```diff
@@ -224,6 +198,63 @@ export class UserFormComponent implements OnInit, OnChanges {
         }
     }
 
+    editOwnAccount() {
+        this.title = this.user.username;
+        this.subtitle = "'appLabels.form.user.title.edit' | translate";
+        this.loadingData = !this.buildForm(this.user);
+    }
+
+    editExistingUser() {
+        this.title = this.name;
+        this.subtitle = "'appLabels.form.user.title.edit' | translate";
+
+        this._dspApiConnection.admin.usersEndpoint
+            .getUserByUsername(this.name)
+            .pipe(take(1))
+            .subscribe(
+                (response: ApiResponseData<UserResponse>) => {
+                    this.user = response.body.user;
+                    this.loadingData = !this.buildForm(this.user);
+                },
+                (error: ApiResponseError) => {
+                    this._errorHandler.showMessage(error);
+                }
+            );
+    }
+
+    defineNewUser() {
+
+        // get existing users to avoid same usernames and email addresses
+        this.allUsers$
+            .pipe(take(1))
+            .subscribe((allUsers) => {
+                for (const user of allUsers) {
+                    // email address of the user should be unique.
+                    // therefore we create a list of existing email addresses to avoid multiple use of user names
+                    this.existingEmails.push(
```

Comment:
existingEmails should be observable which observes existing user emails from the state

Reply:
Very much agree. Let's go for that next week.